exercises/ex3.cpp

```diff
@@ -1,40 +1,32 @@
 /* Could you still use a switch case here? May you can use a map. */
-
 #include <iostream>
+#include <map>
 using namespace std;
 
 int main()
 {
+  map<string, string> famous;
   string textInput;
+  famous["BarackObama"] = "44th president of the United States";
+  famous["SandroPertini"] = "Former President of the Italian Republic";
+  famous["NelsonMandela"] = "Former President of South Africa";
+  famous["MahatmaGandhi"] = "Bapu";
+  famous["DonaldKnuth"] = "Creator of LaTeX";
+  famous["DennisRitchie"] = "Creator of C";
 
   cout << "Enter a famous name+surname, ex. BarackObama " << endl;
   cin >> textInput;
-
-  if (textInput == "BarackObama")
-  {
-    cout << "44th president of the United States" << endl;
-  }
-  else if (textInput == "SandroPertini")
-  {
-    cout << "Former President of the Italian Republic" << endl;
-  }
-  else if (textInput == "NelsonMandela")
-  {
-    cout << "Former President of South Africa" << endl;
-  }
-  else if (textInput == "MahatmaGandhi")
-  {
-    cout << "Bapu" << endl;
-  }
-  else if (textInput == "DonaldKnuth")
-  {
-    cout << "Creator of LaTeX" << endl;
-  }
-  else if (textInput == "DennisRitchie")
+  bool flag = true;
+  for (auto i = famous.begin(); i != famous.end(); ++i)
   {
-    cout << "Creator of C" << endl;
+    if (i->first == textInput)
+    {
+      cout << i->first << " Description: " << i->second << endl;
+      flag = false;
+      break;
+    }
   }
-  else
+  if (flag != false)
   {
     cout << "Invalid input! Please enter a good name!" << endl;
   }
```

Comment:
1. `if (flag != false)` means `if (flag)`, so you could short it
2. you can make it better wrapping this into a function and don't use the flag variable
```c++
void my_magic_name_func(...) {
  ....
  for (auto i = famous.begin(); i != famous.end(); ++i)
  {
    cout << "Creator of C" << endl;
    if (i->first == textInput)
    {
      cout << i->first << " Description: " << i->second << endl;
      return;
    }
  }

  cout << "Invalid input! Please enter a good name!" << endl;
}
```